lib/matplotlib/font_manager.py

```diff
@@ -262,39 +263,38 @@ def OSXInstalledFonts(directories=None, fontext='ttf'):
             files.extend(list_fonts(path, fontext))
     return files
 
-def get_fontconfig_fonts(fontext='ttf'):
+
+@lru_cache()
+def _call_fc_list():
+    """Cache and list the font filenames known to `fc-list`.
     """
-    Grab a list of all the fonts that are being tracked by fontconfig
-    by making a system call to ``fc-list``.  This is an easy way to
-    grab all of the fonts the user wants to be made available to
-    applications, without needing knowing where all of them reside.
+    # Delay the warning by 5s.
+    timer = Timer(5, lambda: warnings.warn(
+        'Matplotlib is building the font cache using fc-list. '
+        'This may take a moment.'))
+    timer.start()
+    try:
+        out = subprocess.check_output(['fc-list', '--format=%{file}'])
+    except (OSError, subprocess.CalledProcessError):
+        return []
+    finally:
+        timer.cancel()
+    fnames = []
+    for fname in out.split(b'\n'):
+        try:
+            fname = six.text_type(fname, sys.getfilesystemencoding())
+        except UnicodeDecodeError:
+            continue
+    return fnames
+
+
+def get_fontconfig_fonts(fontext='ttf'):
+    """List the font filenames known to `fc-list` having the given extension.
     """
     fontext = get_fontext_synonyms(fontext)
+    return [fname for fname in _call_fc_list()
```

Comment:
From a set represented as a {key: 1, key: 1, ...} dict to a list, yes.  I can even document this as a API break :-)